source/AI.cpp

```diff
@@ -3852,7 +3952,12 @@ void AI::IssueOrders(const PlayerInfo &player, const Orders &newOrders, const st
 				existing.type = Orders::HOLD_POSITION;
 
 			hasMismatch |= (existing.type != newOrders.type);
-			hasMismatch |= (existing.target.lock().get() != newTarget);
+			hasMismatch |= (existing.target.lock().get() != targetShip);
+			hasMismatch |= (existing.targetAsteroid.lock().get() != targetAsteroid);
+			// Skip giving any new orders if the fleet is already in harvest mode and the player has selected a new
+			// asteroid.
+			if(hasMismatch && targetAsteroid)
+				alreadyHarvesting |= ((existing.type == newOrders.type) && (newOrders.type == Orders::HARVEST));
```

Comment:
Why is "the player selected an asteroid for their flagship to attack" auto-issuing "escorts should attack this, too!"?

The player fleet should only attack the flagship's target when actually requested, via the "fight my target" command.

Reply:
I need to give you a little bit of background before the short answer.

# Background

Definition of new fleet orders.

| Command | Purpose |
| --- | --- |
| `Orders::HARVEST` | Stay idle.  But if the flagship blows up an asteroid, then proceed to **harvest** flotsam |
| `Orders::MINING` | This is the new "fight my target".  Fleet chases asteroid and blows it up. |

`V` keyboard shortcut targets asteroids (``Orders::MINING`).  It can also toggle harvest behavior (`Orders::HARVEST` or delete orders/idle fleet).

Pressing `V` key will toggle HARVEST on and off if the same asteroid is targeted or if there's no asteroid targeted but the player wants the fleet to harvest (this scenario would occur if player breaks up asteroid and wants fleet to harvest even though they do not have an asteroid scanner).

# Why this logic

Without it, every time a player targets a _new asteroid_ it toggles the harvest behavior.  This would look like a strange bug to a player because they would expect the fleet to just stay in harvest mode if they're targeting a new asteroid.

# Short answer

To avoid this strange "harvest toggling" when choosing new asteroids this logic is necessary.

Reply:
Should this conversation be resolved?